runtime/src/iree/hal/buffer.h

```diff
@@ -426,7 +457,98 @@ enum iree_hal_mapping_mode_bits_t {
 };
 typedef uint32_t iree_hal_mapping_mode_t;
 
-// Implementation-specific mapping data.
+//===----------------------------------------------------------------------===//
+// iree_hal_buffer_params_t
+//===----------------------------------------------------------------------===//
+
+// Parameters defining how a buffer should be allocated.
+//
+// Designed to be zero-initialized: any field with a 0 value will be assigned
+// a default as indicated in the field description.
+//
+// For ergonomics when used from C++ w/o named initializers the first field is
+// the most commonly used so that it can be initialized by location:
+//    some_fn(..., {IREE_HAL_BUFFER_USAGE_FOO}, ...)
+typedef struct iree_hal_buffer_params_t {
```

Comment:
You know, it does make more sense for `iree_hal_buffer_*` structs to be defined in `buffer.h` instead of `allocator.h` ;P